### PR TITLE
Add test for Metal APIs new in iOS 12

### PR DESCRIPTION
**Pull request by moiseev**
These APIs were used in the test for macOS only, but now with the iOS 12
they can also be called on iOS, so let's test them.

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Comment by moiseev** [comment]
@swift-ci Please smoke test Linux platform

**Comment by moiseev** [comment]
@swift-ci Please smoke test Linux platform

File: test/stdlib/Metal.swift [hunk 1]
```diff
@@ -259,10 +259,16 @@ if #available(OSX 10.14, iOS 12.0, tvOS 12.0, *){
       let encoder = cmdBuf.makeRenderCommandEncoder(descriptor: rpDesc)!
       encoder.useResources([buf], usage: MTLResourceUsage.read)
       encoder.useHeaps([heap])
-      #if os(macOS)
+#if os(macOS)
+      encoder.setViewports([MTLViewport()])
+      encoder.setScissorRects([MTLScissorRect(x:0, y:0, width:1, height:1)])
+#elseif os(iOS)
+      if #available(iOS 12.0, *) {
```
Comment (jrose-apple) [inline test/stdlib/Metal.swift]: You can combine these two branches. That's what the `*` in `#available` is for.

**Comment by moiseev** [comment]
@swift-ci Smoke test

**Pull request merged**